### PR TITLE
Support intermediate output

### LINKED ISSUE #57: Cannot fetch intermediate variable
Tested with chainer==5.1.0 and onnx-chainer==1.3.0a1.

```python
import numpy as np

import chainer
import chainer.links as L
import onnx_chainer


class Model(chainer.Chain):

    def __init__(self):
        super().__init__()
        with self.init_scope():
            self.l1 = L.Linear(4)
            self.l2 = L.Linear(5)

    def __call__(self, x):
        y = self.l1(x)
        z = self.l2(y)
        return y, z


model = Model()
x = np.empty((1, 3), dtype=np.float32)
onnx_chainer.export(model, (x,))
```
```
Traceback (most recent call last):
  File "***.py", line 24, in <module>
    onnx_chainer.export(model, (x,))
  File "***/lib/python3.7/site-packages/onnx_chainer/export.py", line 306, in export
    checker.check_model(model)
  File "***/lib/python3.7/site-packages/onnx/checker.py", line 82, in check_model
    C.check_model(model.SerializeToString())
onnx.onnx_cpp2py_export.checker.ValidationError: Nodes in a graph must be topologically sorted, however input 'Input_0' of node: 
input: "Input_0" input: "139839959553304" input: "139839959553528" output: "Gemm_0" op_type: "Gemm" attribute { name: "alpha" f: 1 type: FLOAT } attribute { name: "beta" f: 1 type: FLOAT } attribute { name: "transA" i: 0 type: INT } attribute { name: "transB" i: 1 type: INT }
 is not output of any previous nodes.
```

Note that all of the following conditions worked well.
- Return `z` only.
```python
    def __call__(self, x):
        y = self.l1(x)
        z = self.l2(y)
        return z
```
- Return `y` only.
```python
    def __call__(self, x):
        y = self.l1(x)
        z = self.l2(y)
        return y
```
- `y` is not intermediate variable.
```python
    def __call__(self, x):
        y = self.l1(x)
        z = self.l2(x)
        return y, z
```

### PR DESCRIPTION
**Pull request by disktnk**
- add test case, which outputs intermediate value
- add checker to check all nodes are connected
- add Identity op to handle intermediate output node

Output example:

image

fixes #57

### CODE REVIEW

File: onnx_chainer/export.py [hunk 1]
```diff
@@ -122,12 +122,22 @@ def backward_postprocess(self, function, in_data, out_grad):
...
+        output_names = []
         for o in function.outputs:
             var = o().get_variable_or_none()
             if var is not None:  # If the output is kept
-                self.middle_output_var_to_varnode[id(var)] = id(o())
-
-        output_names = [str(id(o())) for o in function.outputs]
+                output_name = str(id(var))
+                if output_name in self.inputs:
+                    # ONNX graph does not allow one node is both input and
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/export.py]: When one value is shared by graph output and node input, `onnx.checker.check` returns false with 

```
onnx.onnx_cpp2py_export.checker.ValidationError: Graph must be in single static assignment (SSA) form, however '139730786178384' has been used as output names multiple times.
```

However, [IR document](https://github.com/onnx/onnx/blob/master/docs/IR.md#nodes) refers to only outputs SSA. So fixed comments more accurately.

[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/export.py]: fixed

File: onnx_chainer/testing/test_onnxruntime.py [hunk 2]
```diff
@@ -83,23 +83,41 @@ def check_output(model, x, filename, out_keys=None, opset_version=None):
 
     onnx_model = onnx_chainer.export(model, x, filename,
                                      opset_version=opset_version)
+    check_all_connected(onnx_model)
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/testing/test_onnxruntime.py]: fixed

File: onnx_chainer/testing/test_onnxruntime.py [hunk 3]
```diff
@@ -83,23 +83,41 @@ def check_output(model, x, filename, out_keys=None, opset_version=None):
...
         rt_out_keys, {name: array for name, array in zip(input_names, x_rt)})
 
     for cy, my in zip(chainer_out, rt_out):
         np.testing.assert_allclose(cy, my, rtol=1e-5, atol=1e-5)
+
+
+def check_all_connected(onnx_model):
+    edge_names = set(_get_graph_input_names(onnx_model))
+    seen_nodes = set()
+    # ONNX model is directed graph
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/testing/test_onnxruntime.py]: I would intend checking process is insufficient if graph is more complicated such cyclic or un-direct, but the comment is meaningless, so removed. Thx.

File: onnx_chainer/testing/test_onnxruntime.py [hunk 4]
```diff
@@ -83,23 +83,41 @@ def check_output(model, x, filename, out_keys=None, opset_version=None):
...
+    for node in onnx_model.graph.node:
+        for input_name in node.input:
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/testing/test_onnxruntime.py]: refactored.

File: onnx_chainer/export.py [hunk 5]
```diff
@@ -101,8 +101,8 @@ class ONNXExport(chainer.FunctionHook):
     def __init__(self, opset_version=None):
         self.graph = []
         self.inputs = {}
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/export.py]: fixed

File: onnx_chainer/export.py [hunk 6]
```diff
@@ -101,8 +101,8 @@ class ONNXExport(chainer.FunctionHook):
...
+        self.outputs = {}
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/export.py]: fixed

File: onnx_chainer/export.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shinh) [reply onnx_chainer/export.py]: Ah, I haven't realized `onnx.checker` disallows it. Good to know, thanks!

**Pull request merged**